DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.cc

```diff
@@ -189,182 +150,156 @@ void BeamSpotProblemMonitor::analyze(const Event& iEvent,
     return;
   }
 
-    BeamSpotStatus_ = 0.;
-
-
-    // Checking TK status
-    Handle<DcsStatusCollection> dcsStatus;
-    iEvent.getByToken(dcsStatus_, dcsStatus);
-    for (int i=0;i<6;i++) dcsTk[i]=true;
-    for (DcsStatusCollection::const_iterator dcsStatusItr = dcsStatus->begin(); 
-         dcsStatusItr != dcsStatus->end(); ++dcsStatusItr) {
-      if (!dcsStatusItr->ready(DcsStatus::BPIX))   dcsTk[0]=false;
-      if (!dcsStatusItr->ready(DcsStatus::FPIX))   dcsTk[1]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TIBTID)) dcsTk[2]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TOB))    dcsTk[3]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TECp))   dcsTk[4]=false;
-      if (!dcsStatusItr->ready(DcsStatus::TECm))   dcsTk[5]=false;
-    }  
-
-     bool AllTkOn = true;
-        for (int i=0; i<5; i++) 
-            {
-              if (!dcsTk[i]) {
-                              AllTkOn = false;
-                       
-                             }
-            }
-
-
-     //If tracker is ON and collision is going on then must be few track ther
-     edm::Handle<reco::TrackCollection> TrackCollection;
-     iEvent.getByToken(trkSrc_, TrackCollection);
-     const reco::TrackCollection *tracks = TrackCollection.product();
-     for ( reco::TrackCollection::const_iterator track = tracks->begin();track != tracks->end();++track ) 
-      {
-         if(track->pt() > 1.0)Ntracks_++;
-          if(Ntracks_> 200) break;
-      }
+  beamSpotStatus_ = 0.;
+
+
+  // Checking TK status
+  Handle<DcsStatusCollection> dcsStatus;
+  iEvent.getByToken(dcsStatus_, dcsStatus);
+  std::array<bool,6> dcsTk;
+  for (auto &e: dcsTk) { e=true; }
+  for (auto const& status: *dcsStatus ) {
+    if (!status.ready(DcsStatus::BPIX))   dcsTk[0]=false;
+    if (!status.ready(DcsStatus::FPIX))   dcsTk[1]=false;
+    if (!status.ready(DcsStatus::TIBTID)) dcsTk[2]=false;
+    if (!status.ready(DcsStatus::TOB))    dcsTk[3]=false;
+    if (!status.ready(DcsStatus::TECp))   dcsTk[4]=false;
+    if (!status.ready(DcsStatus::TECm))   dcsTk[5]=false;
+  }
+
+  bool allTkOn = true;
+  //BUG??? DcsStatus::TECm is ignored
+  for (int i=0; i<5; i++) {
+    if (!dcsTk[i]) {
```

Comment:
This appears to be a bug from the original version of the code. If you want it fixed, I'd change it to
```
for(auto e: dcsTk) {
   if(!e) {
      allTkOn = false;
      break;
   }
}
```
Which will work even if the size if `dcsTk` is ever changed.

Reply:
@sushilchauhan FYI. You seem to be the author of this code so maybe you would be qualified to answer this question.